src/components/SignUp.js

```diff
@@ -1,7 +1,252 @@
+                  value = {this.state.state}
+                />
+              </Grid>
+              <Grid item xs={12} sm={6}>
+                <TextField
+                  variant="outlined"
+                  fullWidth
+                  id="zipCode"
+                  label="Zip Code"
+                  name="zipCode"
+                  onChange = {this.handleChange}
+                  value = {this.state.zipCode}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  variant="outlined"
+                  fullWidth
+                  id="country"
+                  label="Country"
+                  name="country"
+                  onChange = {this.handleChange}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  variant="outlined"
+                  fullWidth
+                  id="phone"
+                  label="Phone"
+                  name="phone"
+                  onChange = {this.handleChange}
+                  value = {this.state.phone}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  variant="outlined"
+                  required
+                  fullWidth
+                  id="email"
+                  label="Email Address"
+                  name="email"
+                  onChange = {this.handleChange}
+                  value = {this.state.email}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  variant="outlined"
+                  required
+                  fullWidth
+                  name="password"
+                  label="Password"
+                  type="password"
+                  id="password"
+                  onChange = {this.handleChange}
+                  value = {this.state.password}
+                />
+              </Grid>
+            </Grid>
+            <MuiThemeProvider theme={theme}>
+              <Button
+                type="submit"
+                fullWidth
+                variant="contained"
+                color="secondary"
+                className={classes.submit}
+              >
+                Sign Up
+              </Button>
+            </MuiThemeProvider>
+            <Grid container justify="flex-end">
+              <Grid item>
+                <Link href="/login" variant="body2">
+                  Already have an account? Sign in
+                </Link>
+              </Grid>
+            </Grid>
+          </form>
+        </div>
+      </Container>
+    );
+  }
+}
+
+const mapDispatchToProps = dispatch => ({
+  postSignup: (user) => {
+    dispatch(postSignup(user));
+},
+})
+
+const StyledSignUp = withStyles(styles)(SignUp);
+const ConnectedSignUp = connect(null, mapDispatchToProps)(StyledSignUp);
+
+export default ConnectedSignUp;
```

Comment:
We should have a generic form component that we can reuse between login and signup (and maybe, later, adding a beer)